src/log.cr

```diff
@@ -98,25 +115,30 @@
 # Log.setup_from_env
 # ```
 #
-# The environment variables `CRYSTAL_LOG_LEVEL` and `CRYSTAL_LOG_SOURCES` are used to indicate
-# which severity level to emit (defaults to `INFO`; use `NONE` to skip all messages) and to restrict
-# which sources you are interested in.
+# The environment variables `LOG_LEVEL` and `LOG_SOURCES` are used to indicate
+# which severity level to emit. By default entries from all sources with `Info` and above severity will
+# be logged to `STDOUT` using the `Log::IOBackend`.
 #
-# The valid values for `CRYSTAL_LOG_SOURCES` are:
+# To change the level and sources change the environment variables value:
 #
-# * the empty string matches only the top-level source (default)
-# * `*` matches all the sources
+# ```console
+# $ LOG_LEVEL=DEBUG LOG_SOURCES=myapp.* ./bin/app
```

Comment:
I think the more common usecase is "I want info logs for everything, *and* debug logs for `myapp.*`". `LOG_SOURCES` concerns me.

I think `CRYSTAL_LOG_LEVEL=DEBUG` would log everything `> DEBUG`, then `CRYSTAL_LOG_LEVEL="myapp.*=DEBUG"` would leave the default of `*=INFO` in place and override just `myapp`.

Reply:
I like this idea. It's more flexible without too much complications.
I could also approve this PR assuming sources is always `*` now, and add the parsing later. We could support both `LOG_LEVEL=info` and `LOG_LEVEL="*=info, bar=warning". If the source is missing it assumes `*`.

Reply:
My idea was to have something simple and later a file base configuration could enable more fine grained control.

Would it make sense to make in this PR `LOG_LEVEL=<level>` use always `*` as source? And add in a later PR the functionality to parse `<source1>=<level1> <source2>=<level2>` syntax

That would be dropping the `LOG_SOURCES` variable in this PR.

---

Note: With @waj we thought about dropping the `CRYSTAL_` prefix of the variable. It doesn't feel right to force an app to mention the language used under the hood.

Reply:
haha @bcardiff we're in sync

Reply:
We should really try and avoid env var conflicts, `LOG_LEVEL` sounds like it'd easily cause havoc when the crystal program starts spawning subprograms. `CRYSTAL_LOG_LEVEL` is uses crystal-specific syntax for the crystal-specific logging library. I don't think it's a good idea in practice.

I'm 100% ok with dropping `LOG_SOURCES` and adding nicer `CRYSTAL_LOG_LEVEL` parsing in a later PR though.

Reply:
That's not really a valid argument. If the spawned process is built with Crystal you still have a conflict. And if `LOG_LEVEL` causes so much "conflicts" it's because is a broadly accepted name for such purpose. You can still customize the variable name for your apps if you want.